### ISSUE TITLE
Shouldn't report class-variable-slots-conflict for class variables used as typed

### ISSUE THREAD

**Issue by brendanator** [PyCQA/pylint#3141]
### Steps to reproduce
```python
class Example:
  __slots__ = ['field']
  field: int
```
### Current behavior
Reports `class-variable-slots-conflict`

### Expected behavior
Shouldn't report

### pylint --version
pylint 2.4.1
astroid 2.3.0

**Comment by PCManticore** [comment]
Thanks, I agree as well.
